### PR TITLE
Rewritten Export Routine CSV report in the new Rect UI

### PR DESCRIPTION
**Pull request by konavivekramakrishna**
This pr adds the Export Routine CSV report page

fixes : #762 

Before:
image

After:
image

### CODE REVIEW

**Comment by mozzy11** [comment]
Thanks @konavivekramakrishna , 
Basically menu items are added using liquibase changesets to the DB ie [here](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/src/main/resources/liquibase/2.8.x.x/menu.xml#L102-L129) to I order to be compatible with the old jsp UI .,so we have the a single source of  menu items for both the new and old UI

**Comment by mozzy11** [comment]
Which menu item did you need to add ??

**Review by mozzy11 — CHANGES_REQUESTED** [review]
see above comments

**Comment by mozzy11** [comment]
actually, use the 
config.serverBaseUrl + "/RoutineReport?type=patient&report=referredOut"
i renamed it so that all new RoutineReports in the Rect UI can use that 
https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/frontend/src/App.js#L382

**Comment by konavivekramakrishna** [comment]
Okay @mozzy11 

I have another recommendation for the frontend code. How about using Husky as a pre-commit hook to automatically format the frontend code using Prettier? This way, the code remains consistent, and the formatting will be consistent as well. What do you think about it?

Can I create a separate PR for the adding pre commit hook to auto format using prettier?

**Comment by konavivekramakrishna** [comment]
@mozzy11 

I have implemented the changes and attached the screenshot could you please review

File: frontend/src/components/Reports/routine/ReferedOutTestSummary.js
Comment (mherman22) [inline frontend/src/components/Reports/routine/ReferedOutTestSummary.js:1]: i would probably go ahead and name my file as `CISampleRoutineExport.js`

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: frontend/src/components/Reports/Routine.js [hunk 1]
```diff
@@ -82,8 +82,7 @@ export const RoutineReportsMenu = {
       icon: IbmWatsonNaturalLanguageUnderstanding,
       SideNavMenuItem: [
         {
-          link:
-            config.serverBaseUrl + "/Report?type=patient&report=referredOut",
+          link:"/RoutineReport?type=patient&report=referredOut",
```
[the remark this answers is not in the document]
Comment (mherman22) [reply frontend/src/components/Reports/Routine.js:85]: as you can see this `/RoutineReport?type=patient&report=referredOut` is for the referredOut yet this pull request is supposed to address something about ExportRoutineCSV report.

**Comment by konavivekramakrishna** [comment]
@mozzy11 

can you please review

File: frontend/src/components/Reports/Routine.js (hunk 1 above)
Comment (mozzy11) [inline frontend/src/components/Reports/Routine.js:85]: @konavivekramakrishna , 
@mherman22 is right . you worked on the wrong report menu item . 
YOu were supposed to work on [this ](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/frontend/src/components/Reports/Routine.js#L133-L144)  ie it shoulb become 
`"/RoutineReport?type=patient&report=CISampleRoutineExport",`

File: frontend/src/components/Reports/routine/Index.js [hunk 2]
```diff
@@ -88,6 +89,11 @@ const RoutineIndex = () => {
             {type === "indicator" &&
               report === "indicatorCDILNSPHIV" &&
               (<HIVTestSummary/>)}
+            {
+                type === "patient" && 
+                report === "referredOut" &&
```
Comment (mozzy11) [inline frontend/src/components/Reports/routine/Index.js:94]: the reort is supposed to be `CISampleRoutineExport`

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Use the right report menu item

**Pull request closed without merging**